### PR TITLE
docs: migration guide to jina 3

### PR DESCRIPTION
**Pull request by JohannesMessner**
Moved PR due to renamed branch

### CODE REVIEW

File: docs/migrate-to-jina3/migrate.md [hunk 1]
```diff
@@ -0,0 +1,164 @@
+# Migration to Jina 3
+
+Jina 3 introduces a number of exciting features, but to be able to enjoy them, you will also have to make some
+tweaks to your existing Jina 2 code.
+
+One of the major changes in Jina 3 is the inclusion of [docarray](https://docarray.jina.ai/):
+The previously included `Document` and `DocumentArray` data structures now form their own library and include new
+features, improved performance, and increased flexibility.
+Accordingly, the scope of the changes in Jina 3 will be mainly related to `Document` and `DocumentArray`.
```
Comment (alexcg1) [inline docs/migrate-to-jina3/migrate.md:9]: Need a new line here

Comment (alexcg1) [reply docs/migrate-to-jina3/migrate.md:9]: Before `Accordingly`

File: docs/migrate-to-jina3/migrate.md [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+long run. Here you can find out what exactly you will have to adapt.
+
+## More natural attribute names
+
+Docarray introduces more natural naming conventions for `Document` and `DocumentArray` attributes.
+
+- `doc.blob` is renamed to `doc.tensor`, to align with external libraries like PyTorch and Tensorflow
+- `doc.buffer` is renamed to `doc.blob`, to align with the industry standard
+
+## Simplified access of attributes and elements
```
Comment (alexcg1) [inline docs/migrate-to-jina3/migrate.md:21]:
```suggestion
## Simplified access to attributes and elements
```

**Review by alexcg1 — CHANGES_REQUESTED** [review]

**Comment by JohannesMessner** [comment]
Not ready to merge yet, currently incorporating feedback from the doc coaching

File: docs/migrate-to-jina3/migrate.md [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+**Attributes**: Docarray introduces a flexible way of accessing attributes of `Document`s in a `DocumentArray`, in bulk.
+- Instead of having to call `docs.get_attributes('attribute')`, you can simply call `docs.attributes` for
+  a select number of attributes. Currently, this syntax is supported by:
+  - `text`: `docs.texts`
+  - `blob`: `docs.blobs`
+  - `tensor`: `docs.tensors`
+  - `content`: `docs.contents`
+  - `embedding`: `docs.embeddings`
+- The remaining attributes can be accessed in bulk by calling `docs[:, 'attribute']`, e.g. `docs[:, 'tags']`.
+  Additionally, you can access a specific key in `tags` by calling `docs[:'tags__key']`.
```
Comment (JoanFM) [inline docs/migrate-to-jina3/migrate.md:53]:
```suggestion
  Additionally, you can access a specific key in `tags` by calling `docs[:, 'tags__key']`.
```

File: docs/migrate-to-jina3/migrate.md [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+
+docs = DocumentArray([d, d])
+print(docs.texts)
+>>> None
+```
+
+````
+
+
+## Pythonic serialization
```
Comment (JoanFM) [inline docs/migrate-to-jina3/migrate.md:161]: Try to group the changes between `Document`, `DocumentArray` based changes and the `Flow` ones.

**Review by JoanFM — CHANGES_REQUESTED** [review]

**Review by JoanFM — CHANGES_REQUESTED** [review]
Also mention that `DocumentArrayMemmap` is removed

**Review by JoanFM — APPROVED** [review]

**Pull request merged**